docs/guides/Sign_Git_commits.md

```diff
@@ -0,0 +1,120 @@
+# Guide: Sign Git commits
+
+- [Guide: Sign Git commits](#guide-sign-git-commits)
+  - [Overview](#overview)
+  - [Setup](#setup)
+  - [Troubleshooting](#troubleshooting)
+  - [Additional settings](#additional-settings)
+
+## Overview
+
+Signing Git commits is a good practice and ensures the correct web of trust has been established for the distributed version control management, e.g. [Bitwarden](https://bitwarden.com/).
+
+There are two ways to sign commits in GitHub, using a GPG or an SSH signature. Detailed information about this can be found in the following [documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification). It is recommended to use the GPG method for signing commits as GPG keys can be set to expire or be revoked if needed. Below is a step-by-step guide on how to set it up.
+
```

Comment:
```suggestion

## SSH Signing

You should not do this if you already have GPG signing set up. One or the other is fine.  Not both.

If you do not already have SSH key access set up on your Github account, [do that first](https://docs.github.com/en/authentication/connecting-to-github-with-ssh).

After doing that, you will have an authentication key listed in the `SSH and GPG keys` section of your Github profile, and a public and a private key in your `~/.ssh` directory.  We will be using that same key for commit signing in these instructions.

In what follows, I will assume that the private key filename is `~/.ssh/id_rsa`, and the public key filename is `~/.ssh/id_rsa.pub`.

### Configuring Git

The first step is to configure git to use SSH keysigning, rather than GPG.  Run the following:

    git config --global gpg.format ssh

Next, you need to tell git which private key to use for signing:

    git config --global user.signingkey ~/.ssh/id_rsa

Next, you need to tell git that you want to sign all your commits.  This isn't strictly necessary, but the alternative is adding `-S` on every `git commit`, and that's annoying.  Run:

    git config --global commit.gpgsign true

### Configuring Github

Upload `~/.ssh/id_rsa.pub` to your GitHub profile into the [SSH keys](https://github.com/settings/keys) section *being sure to select `Signing key` from the `Key type` dropdown*.  When it's uploaded, because you're using the same key for both authentication and for signing, on the `SSH and GPG keys` page, you should see that the signing key has the same signature as your authentication key.

### Testing

Make a commit to a branch locally, and push it to Github.  When you view the branch history on Github, you should now see that your latest commit has a `Verified` tag.

```

Reply:
I've put this bit here because the last line of the file is a code fence, which completely breaks the github `suggestion` mechanism.  SSH and GPG could go in either order.

Reply:
done, thanks